trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/MainDispatcher.java

```diff
@@ -424,8 +409,21 @@ public Result dispatch(final IASTNode n) {
 			final ILocation loc = mLocationFactory.createCLocation(n);
 			throw new UnsupportedSyntaxException(loc, msg);
 		}
-		handleWitnessInvariantsAfterDispatch(n, result);
-		return result;
+		return handleWitnessEntries(n, result);
+	}
+
+	public Result handleWitnessEntries(final IASTNode node, final Result result) {
+		if (mWitnessEntries == null) {
+			return result;
+		}
+		Result rtr = result;
+		final ILocation loc = mLocationFactory.createCLocation(node);
+		for (final IExtractedWitnessEntry entry : mWitnessEntries.getImage(node)) {
+			if (mNodeLabelsOfAddedWitnesses.add(entry.getNodeLabels())) {

```

Comment:
nitpick: side-effects in `if` conditions are not so nice; moving it into a separate line (and storing the result in a boolean variable) would be better

Reply:
I disagree for this particular case. Adding something to a container and checking if it was already present is IMHO nicer if its done like that.

Reply:
For me, storing that result in a nicely named variable also helps me to remember the meaning of the boolean returned by `add()`